### ISSUE TITLE
Error in Training

### ISSUE THREAD

**Issue by CPFLAME** [Engineering-Course/LIP_SSL#1]
It's an amazing work, I just follow your step, and your pretrained model performs quite well in the Testset.
But when I start training in training and val set by finetuning your pretrained model, I met this error:

I0516 17:22:28.534246 34577 layer_factory.hpp:77] Creating layer data
I0516 17:22:28.534312 34577 net.cpp:106] Creating Layer data
I0516 17:22:28.534332 34577 net.cpp:411] data -> data
I0516 17:22:28.534373 34577 net.cpp:411] data -> label
I0516 17:22:28.534395 34577 net.cpp:411] data -> (automatic)
I0516 17:22:28.534835 34577 image_seg_data_layer.cpp:46] Opening file human/list/train.txt
I0516 17:22:28.566068 34577 image_seg_data_layer.cpp:63] Shuffling data
I0516 17:22:28.569484 34577 image_seg_data_layer.cpp:68] A total of 40462 images.
I0516 17:22:28.584575 34577 image_seg_data_layer.cpp:137] output data size: 1,3,321,321
I0516 17:22:28.584602 34577 image_seg_data_layer.cpp:141] output label size: 1,1,321,321
I0516 17:22:28.584614 34577 image_seg_data_layer.cpp:145] output data_dim size: 1,1,1,2
I0516 17:22:28.594056 34577 net.cpp:150] Setting up data
I0516 17:22:28.594171 34577 net.cpp:157] Top shape: 1 3 321 321 (309123)
I0516 17:22:28.594243 34577 net.cpp:157] Top shape: 1 1 321 321 (103041)
I0516 17:22:28.594271 34577 net.cpp:157] Top shape: 1 1 1 2 (2)
I0516 17:22:28.594316 34577 net.cpp:165] Memory required for data: 1648664
I0516 17:22:28.594347 34577 layer_factory.hpp:77] Creating layer data_data_0_split
I0516 17:22:28.594403 34577 net.cpp:106] Creating Layer data_data_0_split
I0516 17:22:28.594442 34577 net.cpp:454] data_data_0_split <- data
I0516 17:22:28.594481 34577 net.cpp:411] data_data_0_split -> data_data_0_split_0
I0516 17:22:28.594521 34577 net.cpp:411] data_data_0_split -> data_data_0_split_1
I0516 17:22:28.594554 34577 net.cpp:411] data_data_0_split -> data_data_0_split_2
I0516 17:22:28.594708 34577 net.cpp:150] Setting up data_data_0_split
I0516 17:22:28.594743 34577 net.cpp:157] Top shape: 1 3 321 321 (309123)
I0516 17:22:28.594769 34577 net.cpp:157] Top shape: 1 3 321 321 (309123)
I0516 17:22:28.594792 34577 net.cpp:157] Top shape: 1 3 321 321 (309123)
I0516 17:22:28.594813 34577 net.cpp:165] Memory required for data: 5358140
I0516 17:22:28.594836 34577 layer_factory.hpp:77] Creating layer label_data_1_split
I0516 17:22:28.594862 34577 net.cpp:106] Creating Layer label_data_1_split
I0516 17:22:28.594884 34577 net.cpp:454] label_data_1_split <- label
I0516 17:22:28.594910 34577 net.cpp:411] label_data_1_split -> label_data_1_split_0
I0516 17:22:28.594944 34577 net.cpp:411] label_data_1_split -> label_data_1_split_1
I0516 17:22:28.595024 34577 net.cpp:150] Setting up label_data_1_split
I0516 17:22:28.595055 34577 net.cpp:157] Top shape: 1 1 321 321 (103041)
I0516 17:22:28.595078 34577 net.cpp:157] Top shape: 1 1 321 321 (103041)
I0516 17:22:28.595098 34577 net.cpp:165] Memory required for data: 6182468
I0516 17:22:28.595119 34577 layer_factory.hpp:77] Creating layer shrink_data05
I0516 17:22:28.595193 34577 net.cpp:106] Creating Layer shrink_data05
I0516 17:22:28.595221 34577 net.cpp:454] shrink_data05 <- data_data_0_split_0
I0516 17:22:28.595247 34577 net.cpp:411] shrink_data05 -> shrink_data05
I0516 17:22:28.595319 34577 net.cpp:150] Setting up shrink_data05
I0516 17:22:28.595353 34577 net.cpp:157] Top shape: 1 3 161 161 (77763)
I0516 17:22:28.595376 34577 net.cpp:165] Memory required for data: 6493520
I0516 17:22:28.595396 34577 layer_factory.hpp:77] Creating layer shrink_data075
I0516 17:22:28.595424 34577 net.cpp:106] Creating Layer shrink_data075
I0516 17:22:28.595445 34577 net.cpp:454] shrink_data075 <- data_data_0_split_1
I0516 17:22:28.595471 34577 net.cpp:411] shrink_data075 -> shrink_data075
I0516 17:22:28.595533 34577 net.cpp:150] Setting up shrink_data075
I0516 17:22:28.595563 34577 net.cpp:157] Top shape: 1 3 241 241 (174243)
I0516 17:22:28.595584 34577 net.cpp:165] Memory required for data: 7190492
I0516 17:22:28.595628 34577 layer_factory.hpp:77] Creating layer conv1_1
I0516 17:22:28.595676 34577 net.cpp:106] Creating Layer conv1_1
I0516 17:22:28.595702 34577 net.cpp:454] conv1_1 <- data_data_0_split_2
I0516 17:22:28.595729 34577 net.cpp:411] conv1_1 -> conv1_1
E0516 17:22:28.599076 34580 io.cpp:81] Could not open or find file human/data/
F0516 17:22:28.599761 34580 data_transformer.cpp:475] Check failed: img_height == seg_height (246 vs. 0) 
*** Check failure stack trace: ***
    @     0x7f6489caddaa  (unknown)
    @     0x7f6489cadce4  (unknown)
    @     0x7f6489cad6e6  (unknown)
    @     0x7f6489cb0687  (unknown)
    @     0x7f648a31d77d  caffe::DataTransformer<>::TransformImgAndSeg()
    @     0x7f648a442c51  caffe::ImageSegDataLayer<>::load_batch()
    @     0x7f648a453acc  caffe::BasePrefetchingDataLayer<>::InternalThreadEntry()
    @     0x7f648a351d80  caffe::InternalThread::entry()
    @     0x7f64818afa4a  (unknown)
    @     0x7f647d414184  start_thread
    @     0x7f648908a37d  (unknown)
    @              (nil)  (unknown)

Any advice would be nice : )

**Comment by Engineering-Course** [comment]
E0516 17:22:28.599076 34580 io.cpp:81] Could not open or find file human/data/

Please check your directory of the image data.

**Comment by CPFLAME** [comment]
Thank you for your reply.
my train.txt is the same as test.txt,it is just like this:
![image](https://cloud.githubusercontent.com/assets/13843079/26135231/d23be976-3ae6-11e7-955d-d10005b8b1e4.png)
And my files in human/data:
![image](https://cloud.githubusercontent.com/assets/13843079/26135355/8ddb9ae6-3ae7-11e7-978a-c70099e0b030.png)
I'm so confused that I can run TEST but I failed when training.
I run the code in the root path :  LIP_SSL

**Comment by Engineering-Course** [comment]
Oh, train.txt should contain both image and label split by blank, which is different from test.txt.

**Comment by CPFLAME** [comment]
Thanks for your reply. It works fine now!

**Comment by staywithme23** [comment]
@CPFLAME May I know if you uncomment USE_CUDA when make caffe folder?
I ran into #2 problem.
Best,

**Comment by CPFLAME** [comment]
@adwin5 I used cuda when I make caffe folder.
It seems that the memory of your GPU is not big enough to run test, I think you can choose a GPU of larger memory to run this code. 
By the way, I used the memory of 2579M  to run the test code. You can  check your momery size.

**Comment by staywithme23** [comment]
@CPFLAME Thank you!
I will try uncomment USE_CUDA
I have 8G, should be OK for testing!

**Comment by Engineering-Course** [comment]
@adwin5 You may use cuDNN or a larger memory GPU.

**Comment by staywithme23** [comment]
@Engineering-Course 
Thanks it works.

**Comment by staywithme23** [comment]
@Engineering-Course @CPFLAME 
I can run training, and I reference init.model from deeplap attention model.
I want to ask how to set path to use pose ground truth for training.
The run_human.sh code seems just reading ground truth labels not including pose, right?

My second question is what does strong and weak means in the training section in the run_human.sh file
Thank you.

**Comment by Engineering-Course** [comment]
@adwin5 
Sorry, these scripts are only used for parsing tasks. 
Strong and weak mean nothing which are used by deeplab not this project.

**Comment by staywithme23** [comment]
@Engineering-Course 
Can I confirm with you that the code is just using parsing loss (pixel-wise softmax loss) instead of structure loss (structure-sensitive loss) which mentioned in paper.
Thank you.

**Comment by Engineering-Course** [comment]
@adwin5 NONONO, it does use a structure loss. You can see the configuration in train.prototxt.
